src/graphql/utils/buildObject.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * A utility function for easily building an object from lists *without*
+ * tolerating duplicate keys. If any duplicate keys are seen an error will be
+ * thrown.
+ *
+ * This is useful for when we are building map objects as is often done for
+ * GraphQL configs.
+ *
+ * This is pretty much identical to the Lodash `_.fromPairs` function.
+ *
+ * ## Example:
+ * ```js
+ * const actual = buildObject(
+ *   [
+ *     ['a', 1],
+ *     ['b', 2],
+ *   ],
+ *   [
+ *     ['c', 3],
+ *   ]
+ * )
+ *
+ * const expected = {
+ *   a: 1,
+ *   b: 2,
+ *   c: 3,
+ * }
+ *
+ * assert.deepEqual(actual, expected)
+ * ```
+ */
+export default function buildObject <T>(
+  ...entriess: Array<Array<[string, T] | undefined | null | false>>
+): { [key: string]: T } {
+  const object = {}
+
+  entriess.forEach(entries => entries.forEach(entry => {
+    if (!entry)
+      return
+
+    const [key, value] = entry
+
+    if (object.hasOwnProperty(key))
+      throw new Error(`Naming conflict when building object. Cannot have two definitions for property '${key}'.`)
```

Comment:
`fooBefore` and `myFunctionNameAfter` more closely follows the SQL pattern of `mytablename_mycolumnname` for naming columns. And if `foo_before` and `before` columns both exists on a table, then `before` can before `fooBefore2`.

I'm not sure what happens if `before` is processed in the for loop before `foo_before`. So maybe that won't work. Something is going to have to keep the map order consistent.

All of these tell me that the column is on the table `foo`. When I see `before_`, I'm confused. Is that something private? Is it a concatenation bug? If we're camel casing properties, what is `before_` going to be?